executor/executor.cc

```diff
@@ -1254,9 +1255,18 @@ void execute_call(thread_t* th)
 		cover_reset(&th->cov);
 	// For pseudo-syscalls and user-space functions NONFAILING can abort before assigning to th->res.
 	// Arrange for res = -1 and errno = EFAULT result for such case.
-	th->res = -1;
+	res = -1;
 	errno = EFAULT;
-	NONFAILING(th->res = execute_syscall(call, th->args));
+	NONFAILING(res = execute_syscall(call, th->args));
+#if SYZ_HAVE_DOEXIT_THREAD
+	if (call->attrs.forks && res == 0) {
```

Comment:
I think you need to add the same check in the loop that reruns a system call:
```
	for (int i = 0; i < th->call_props.rerun; i++)
		NONFAILING(execute_syscall(call, th->args));
```

Reply:
Even with this fix, I see cases when syzkaller makes too many processes. It can be due to clone(CLONE_VM, stack = 0, ...). In this case, child and parent processes runs on the same stack, so the child can see a wrong value in res.